pkg/ca/extensions.go

```diff
@@ -15,83 +15,9 @@
 package ca
 
 import (
-	"crypto/x509/pkix"
-	"encoding/asn1"
-	"errors"
+	"github.com/sigstore/fulcio/pkg/certificate"
 )
 
 // Extensions contains all custom x509 extensions defined by Fulcio
-type Extensions struct {
-	// NB: New extensions must be added here and documented
-	// at docs/oidc-info.md
-
-	// The OIDC issuer. Should match `iss` claim of ID token or, in the case of
-	// a federated login like Dex it should match the issuer URL of the
-	// upstream issuer. The issuer is not set the extensions are invalid and
-	// will fail to render.
-	Issuer string // OID 1.3.6.1.4.1.57264.1.1
-
-	// Triggering event of the Github Workflow. Matches the `event_name` claim of ID
-	// tokens from Github Actions
-	GithubWorkflowTrigger string // OID 1.3.6.1.4.1.57264.1.2
-
-	// SHA of git commit being built in Github Actions. Matches the `sha` claim of ID
-	// tokens from Github Actions
-	GithubWorkflowSHA string // OID 1.3.6.1.4.1.57264.1.3
-
-	// Name of Github Actions Workflow. Matches the `workflow` claim of the ID
-	// tokens from Github Actions
-	GithubWorkflowName string // OID 1.3.6.1.4.1.57264.1.4
-
-	// Repository of the Github Actions Workflow. Matches the `repository` claim of the ID
-	// tokens from Github Actions
-	GithubWorkflowRepository string // OID 1.3.6.1.4.1.57264.1.5
-
-	// Git Ref of the Github Actions Workflow. Matches the `ref` claim of the ID tokens
-	// from Github Actions
-	GithubWorkflowRef string // 1.3.6.1.4.1.57264.1.6
-}
-
-func (e Extensions) Render() ([]pkix.Extension, error) {
-	var exts []pkix.Extension
-
-	if e.Issuer != "" {
-		exts = append(exts, pkix.Extension{
-			Id:    asn1.ObjectIdentifier{1, 3, 6, 1, 4, 1, 57264, 1, 1},
-			Value: []byte(e.Issuer),
-		})
-	} else {
-		return nil, errors.New("extensions must have a non-empty issuer url")
-	}
-	if e.GithubWorkflowTrigger != "" {
-		exts = append(exts, pkix.Extension{
-			Id:    asn1.ObjectIdentifier{1, 3, 6, 1, 4, 1, 57264, 1, 2},
-			Value: []byte(e.GithubWorkflowTrigger),
-		})
-	}
-	if e.GithubWorkflowSHA != "" {
-		exts = append(exts, pkix.Extension{
-			Id:    asn1.ObjectIdentifier{1, 3, 6, 1, 4, 1, 57264, 1, 3},
-			Value: []byte(e.GithubWorkflowSHA),
-		})
-	}
-	if e.GithubWorkflowName != "" {
-		exts = append(exts, pkix.Extension{
-			Id:    asn1.ObjectIdentifier{1, 3, 6, 1, 4, 1, 57264, 1, 4},
-			Value: []byte(e.GithubWorkflowName),
-		})
-	}
-	if e.GithubWorkflowRepository != "" {
-		exts = append(exts, pkix.Extension{
-			Id:    asn1.ObjectIdentifier{1, 3, 6, 1, 4, 1, 57264, 1, 5},
-			Value: []byte(e.GithubWorkflowRepository),
-		})
-	}
-	if e.GithubWorkflowRef != "" {
-		exts = append(exts, pkix.Extension{
-			Id:    asn1.ObjectIdentifier{1, 3, 6, 1, 4, 1, 57264, 1, 6},
-			Value: []byte(e.GithubWorkflowRef),
-		})
-	}
-	return exts, nil
-}
+// Deprecated: use certificate.Extensions instead.
+type Extensions = certificate.Extensions
```

Comment:
Thoughts on just dropping this? Definitely the right thing to do to avoid breaking clients, but I see no usage outside of fulcio - https://cs.github.com/?scopeName=All+repos&scope=&q=sigstore+AND+ca.Extensions

Reply:
Fine by me! Done.